### ISSUE TITLE
Wrong output for broadcast_in_dim

### ISSUE THREAD

**Issue by hawkinsp** [google/jax#16420]
### Description

Expected behavior, from the CPU backend:
```
In [1]: import jax, jax.numpy as jnp

In [2]: x = jnp.arange(2)

In [3]: jax.lax.broadcast_in_dim(x, (2, 2), (0,))
Out[3]:
Array([[0, 0],
       [1, 1]], dtype=int32)

In [4]: jax.lax.broadcast_in_dim(x, (2, 2), (1,))
Out[4]:
Array([[0, 1],
       [0, 1]], dtype=int32)
```

Actual behavior:
```
In [1]: import jax, jax.numpy as jnp

In [2]: x = jnp.arange(2)
Metal device set to: Apple M1 Pro

systemMemory: 32.00 GB
maxCacheSize: 10.67 GB


In [3]: jax.lax.broadcast_in_dim(x, (2, 2), (0,))
Out[3]:
Array([[0, 1],
       [0, 1]], dtype=int32)

In [4]: jax.lax.broadcast_in_dim(x, (2, 2), (1,))
Out[4]:
Array([[0, 1],
       [0, 1]], dtype=int32)
```

It looks like we're ignoring the dimension argument.

### What jax/jaxlib version are you using?

jax 0.4.11, jaxlib 0.4.10, jax-metal 0.0.2

### Which accelerator(s) are you using?

Apple GPU

**Comment by kulinseth** [comment]
@hawkinsp , can you please check. This has been fixed with jax-metal==0.3.0

**Comment by kulinseth** [comment]
Please re-open, if the issue persists.
